loadfile/loadfile_test.go

```diff
@@ -127,14 +127,14 @@ func Test_MergeFileCaches(t *testing.T) {
 		filename1:  content1,
 		commonName: content,
 	}
-	rootDir1 := "1"
-	fc1 := loadfile.NewFileCache(rootDir1, files1)
+	expRootDir := "1"
+	fc1 := loadfile.NewFileCache(expRootDir, files1)
 
 	files2 := map[string][]byte{
 		commonName: replacedContent,
 		filename2:  content2,
 	}
-	expRootDir := "2"
+
```

Comment:
There ought to also be a unit test where the roots *are* different, expecting the error.

Reply:
Since this PR is intended to resolve a blocked "customer", I decided against requesting this...in the hopes that this will all be refactored and the appropriate testing will appear as part of that work.

Reply:
Fair enough. You "decided against requesting [this]" in several instance with pre-resolved comments, though, so I'll just (post-)pre-resolve this one, too. 😆

Reply:
I added a unit test for this